src/main/resources/com/google/api/codegen/java/settings.snip

```diff
@@ -116,310 +78,356 @@
-                for (RequestIssuer<{@inTypeName}, {@outTypeName}> responder : bundle) {
-                  responder.setException(throwable);
-                }
-              }
+            {@desc.requestTypeName} bundleRequest =
+                {@desc.requestTypeName}.newBuilder()
+                    {@fieldCopyCalls(desc.discriminatorFieldCopies)}
+                    .{@desc.bundledFieldSetFunction}(elements)
+                    .build();
+            return bundleRequest;
+          }
 
-              @@Override
-              public long countElements({@inTypeName} request) {
-                return request.{@getCountCallName(bundling.getBundledField)}();
+          @@Override
+          public void splitResponse(
+              {@desc.responseTypeName} bundleResponse,
+              Collection<? extends RequestIssuer<{@desc.requestTypeName}, {@desc.responseTypeName}>> bundle) {
+            # TODO(garrettjones) support case of no subresponse_field (e.g. Logging)
+            int bundleMessageIndex = 0;
+            for (RequestIssuer<{@desc.requestTypeName}, {@desc.responseTypeName}> responder : bundle) {
+              {@desc.subresponseTypeName} subresponseElements = new ArrayList<>();
+              int subresponseCount = responder.getRequest().{@desc.bundledFieldCountGetFunction}();
+              for (int i = 0; i < subresponseCount; i++) {
+                subresponseElements.add(bundleResponse.{@desc.subresponseByIndexGetFunction}(bundleMessageIndex));
+                bundleMessageIndex += 1;
               }
+              {@desc.responseTypeName} response =
+                  {@desc.responseTypeName}.newBuilder().{@desc.subresponseSetFunction}(subresponseElements).build();
+              responder.setResponse(response);
+            }
+          }
 
-              @@Override
-              public long countBytes({@inTypeName} request) {
-                return request.getSerializedSize();
-              }
-            };
+          @@Override
+          public void splitException(
+              Throwable throwable,
+              Collection<? extends RequestIssuer<{@desc.requestTypeName}, {@desc.responseTypeName}>> bundle) {
+            for (RequestIssuer<{@desc.requestTypeName}, {@desc.responseTypeName}> responder : bundle) {
+              responder.setException(throwable);
+            }
+          }
 
-      @end
-    @end
+          @@Override
+          public long countElements({@desc.requestTypeName} request) {
+            return request.{@desc.bundledFieldCountGetFunction}();
+          }
+
+          @@Override
+          public long countBytes({@desc.requestTypeName} request) {
+            return request.getSerializedSize();
+          }
+        };
+    {@""}
   @end
 @end
 
-@private partitionKeyCode(bundling)
-  {@context.partitionKeyCode(bundling.getDiscriminatorFields)}
+@private partitionKeyCode(bundlingDesc)
+  @join partitionKey : bundlingDesc.partitionKeys on " + "
```

Comment:
Hmmm... you could create a new function in the snippet that returned partitionKey.separatorLiteral with the plus signs on each end, and use that in the join statement. But now the whole thing is feeling convoluted, so I am happy to drop it.